### PR TITLE
NVyrsta

### PR DESCRIPTION
**Pull request by NVyrsta**
[DEMO LINK](https://NVyrsta.github.io/react_movies-list-add-form/)

### CODE REVIEW

File: src/components/NewMovie/NewMovie.tsx [hunk 1]
```diff
@@ -1,45 +1,98 @@
...
         required
       />
 
       <TextField
         name="description"
         label="Description"
-        value=""
+        value={description}
+        onChange={setDescription}
+        required
```
Comment (nzrii-kysel) [inline src/components/NewMovie/NewMovie.tsx:71]: that's why your tests are failing
```suggestion
```

File: src/components/NewMovie/NewMovie.tsx [hunk 2]
```diff
@@ -1,45 +1,98 @@
...
+    callbacks.forEach(callback => callback(''));
+  };
+
+  const newMovie: Movie = {
+    title,
+    description,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  };
```
Comment (nzrii-kysel) [inline src/components/NewMovie/NewMovie.tsx:42]: you shouldn't create `newMovie` on each render, create it before adding it to the array

File: src/components/NewMovie/NewMovie.tsx [hunk 3]
```diff
@@ -1,45 +1,98 @@
...
+
+  const handleSubmit = (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (isFormFilled) {
+      onAdd(newMovie);
+      setCount(prevCount => prevCount + 1);
```
Comment (nzrii-kysel) [inline src/components/NewMovie/NewMovie.tsx:49]: i'd move this `setCount` call into `clearForm`

File: src/components/NewMovie/NewMovie.tsx [hunk 4]
```diff
@@ -1,45 +1,98 @@
...
+  ];
+
+  const isFormFilled = inputs
+    .map(input => input.trim())
+    .every(Boolean);
+
+  const clearForm = () => {
+    callbacks.forEach(callback => callback(''));
+  };
+
```
Comment (nzrii-kysel) [inline src/components/NewMovie/NewMovie.tsx:35]: some things I would do to improve the code
```suggestion
  const requiredInputs = [title, imgUrl, imdbUrl, imdbId];

  const setters = [
    setTitle,
    setDescription,
    setImgUrl,
    setImdbId,
    setImdbUrl,
  ];

  const isFormFilled = requiredInputs.every(input => !!input.trim())

  const clearForm = () => {
    setters.forEach(setter => setter(''));
  };

```

File: src/components/NewMovie/NewMovie.tsx [hunk 5]
```diff
@@ -1,45 +1,98 @@
...
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = (props) => {
+  const { onAdd } = props;
```
Comment (nzrii-kysel) [inline src/components/NewMovie/NewMovie.tsx:10]:
```suggestion
export const NewMovie: React.FC<Props> = ({ onAdd }) => {
```

**Review by nzrii-kysel — APPROVED** [review]

File: src/components/NewMovie/NewMovie.tsx (hunk 5 above)
Comment (NVyrsta) [reply src/components/NewMovie/NewMovie.tsx]: @nzrii-kysel thank you so much, so many useful tips in one review😀

File: src/components/NewMovie/NewMovie.tsx [hunk 6]
```diff
@@ -1,45 +1,92 @@
...
+    setImgUrl,
+    setImdbId,
+    setImdbUrl,
+  ];
+
+  const isFormFilled = requiredInputs.every(input => !!input.trim());
+
+  const clearForm = () => {
+    setCount(prevCount => prevCount + 1);
+    setters.forEach(setter => setter(''));
```
Comment (alexandra-protyanova) [inline src/components/NewMovie/NewMovie.tsx:31]: 🔥

**Review by alexandra-protyanova — APPROVED** [review]
GJ!

**Pull request closed without merging**